xray_core/src/buffer_view.rs

```diff
@@ -271,11 +271,11 @@ impl BufferView {
     }
 
     pub fn set_cursor_position(&mut self, position: Point, autoscroll: bool) {
+        let clipped_point = self.buffer.borrow().clip_point(position);
```

Comment:
Would you be cool just shadowing the original name `position`?

Reply:
Sure

Reply:
And what test coverage should be added?